src/libopensc/card-openpgp.c

```diff
@@ -98,6 +98,19 @@ static pgp_ec_curves_t	ec_curves_openpgp[] = {
 	{{{-1}}, 0} /* This entry must not be touched. */
 };
 
+/* v3.0+ supports: [RFC 4880 & 6637] 0x12 = ECDH, 0x13 = ECDSA */
+static pgp_ec_curves_t  ec_curves_openpgp34[] = {
+        {{{1, 2, 840, 10045, 3, 1, 7, -1}}, 256}, /* ansiX9p256r1 */
+        {{{1, 3, 132, 0, 34, -1}}, 384}, /* ansiX9p384r1 */
+        {{{1, 3, 132, 0, 35, -1}}, 521}, /* ansiX9p521r1 */
+        {{{1, 3, 36, 3, 3, 2, 8, 1, 1, 7, -1}}, 256}, /* brainpoolP256r1 */
+        {{{1, 3, 36, 3, 3, 2, 8, 1, 1, 11, -1}}, 384}, /* brainpoolP384r1 */
+        {{{1, 3, 36, 3, 3, 2, 8, 1, 1, 13, -1}}, 512}, /* brainpoolP512r1 */
+        {{{1, 3, 6, 1, 4, 1, 3029, 1, 5, 1, -1}}, 256}, /* curve25519 for encryption => CKK_EC_MONTGOMERY */
+        {{{1, 3, 6, 1, 4, 1, 11591, 15, 1, -1}}, 256}, /* ed25519 for signatures => CKK_EC_EDWARDS */
+        {{{-1}}, 0} /* This entry must not be touched. */
+};
+
```

Comment:
This is the same as the original `ec_curves_openpgp` aside of the curve25519 and ed25519 addition so I am wondering if we could simplify this somehow to avoid list duplication. Either by prepending these new to the original list and using `ec_curves_openpgp` as a pointer to the third element or some other creative (but readable) way?

Reply:
Thank you! I'm changing it to be similar to a pgp_objects